embedding/csharp/Tizen.Flutter.Embedding/Channels/StandardMethodCodec.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        public static StandardMethodCodec Instance => new StandardMethodCodec(StandardMessageCodec.Instance);
+
+        /// <summary>
+        /// A <see cref="StandardMessageCodec"/>.
+        /// </summary>
+        public StandardMessageCodec MessageCodec { get; private set; }
+
+        /// <InheritDoc/>
+        public byte[] EncodeMethodCall(MethodCall methodCall)
+        {
+            using (var stream = new MemoryStream())
+            using (var writer = new BinaryWriter(stream))
+            {
+                MessageCodec.WriteValue(writer, methodCall.Method);
+                MessageCodec.WriteValue(writer, methodCall.Arguments);
+                return stream.ToArray();
+            }
+        }
+
+        /// <InheritDoc/>
+        public MethodCall DecodeMethodCall(byte[] methodCall)
+        {
+            using (var stream = new MemoryStream(methodCall))
+            using (var reader = new BinaryReader(stream))
+            {
+                object method = MessageCodec.ReadValue(reader);
+                object arguments = MessageCodec.ReadValue(reader);
+                if (method is String strMethod && !stream.HasRemaining())
+                {
+                    return new MethodCall(strMethod, arguments);
+                }
+                throw new ArgumentException("Method call corrupted");
+            }
+        }
+
+        /// <InheritDoc/>
+        public byte[] EncodeSuccessEnvelope(object result)
+        {
+            using (var stream = new MemoryStream())
+            using (var writer = new BinaryWriter(stream))
+            {
+                stream.WriteByte(0);
+                MessageCodec.WriteValue(writer, result);
+                return stream.ToArray();
+            }
+        }
+
+        /// <InheritDoc/>
+        public byte[] EncodeErrorEnvelope(string errorCode, string errorMessage, object errorDetails)
+        {
+            return EncodeErrorEnvelope(errorCode, errorMessage, errorDetails, null);
+        }
+
+        /// <InheritDoc/>
+        public byte[] EncodeErrorEnvelope(string errorCode, string errorMessage, object errorDetails, String errorStacktrace)
+        {
+            using (var stream = new MemoryStream())
+            using (var writer = new BinaryWriter(stream))
+            {
+                stream.WriteByte(1);
+                MessageCodec.WriteValue(writer, errorCode);
+                MessageCodec.WriteValue(writer, errorMessage);
+                if (errorDetails is Exception exception)
+                {
+                    MessageCodec.WriteValue(writer, exception.StackTrace);
+                }
+                else
+                {
+                    MessageCodec.WriteValue(writer, errorDetails);
+                }
+                if (!string.IsNullOrEmpty(errorStacktrace))
+                {
+                    MessageCodec.WriteValue(writer, errorStacktrace);
+                }
```

Comment:
Is it okay if `errorDetails` is null? As far as I can see, the only possible cases include
- both `errorDetails` and `errorStacktrace` are null
- `errorDetails` is null and `errorStacktrace` is non-null
- `errorDetails` is non-null and `errorStacktrace` is null

Reply:
It's okay. If the `errorDetails` is null, the zero(`0`) value as a `NULL` identifier will be written well.